public/pages/CreateTrigger/containers/DefineTrigger/utils/validation.js

```diff
@@ -4,27 +4,22 @@
  */
 import _ from 'lodash';
 
-import { FORMIK_INITIAL_TRIGGER_VALUES, TRIGGER_TYPE } from '../../CreateTrigger/utils/constants';
-
-export const validateTriggerName = (triggers, triggerToEdit, fieldPath, isFullText) => (value) => {
-  if (!value) return isFullText ? 'Trigger name is required.' : 'Required.';
-  const nameExists = triggers.filter((trigger) => {
-    const triggerId = _.get(
-      trigger,
-      `${TRIGGER_TYPE.BUCKET_LEVEL}.id`,
-      _.get(trigger, `${TRIGGER_TYPE.QUERY_LEVEL}.id`)
-    );
-    const triggerName = _.get(
-      trigger,
-      `${TRIGGER_TYPE.BUCKET_LEVEL}.name`,
-      _.get(trigger, `${TRIGGER_TYPE.QUERY_LEVEL}.name`, FORMIK_INITIAL_TRIGGER_VALUES.name)
-    );
-    const triggerToEditId = _.get(triggerToEdit, `${fieldPath}id`, triggerToEdit.id);
-    return triggerToEditId !== triggerId && triggerName.toLowerCase() === value.toLowerCase();
-  });
-  if (nameExists.length > 0) {
-    return 'Trigger name already used.';
-  }
-  // TODO: character restrictions
-  // TODO: character limits
+export const validateTriggerName = (triggers, index) => {
```

Comment:
I see when calling the `validateTriggerName` function, you pass in `triggerValues?.triggerDefinitions` as the `triggers` parameter. This suggests `triggers` could potentially be `undefined` if `triggerValues` is also `undefined`. Would it be worthwhile to assign `triggers` a default value?
E.g.,
```
validateTriggerName = (triggers = [], index) => ...

Reply:
sounds a good idea, will address.